### PR TITLE
Fix the situation where get_parent_tfvars() is called from within the…

### PR DESCRIPTION
**Pull request by jocgir**
… child configuration file. Also apply to path_relative_from_include().

Add test case.

Solve issue #332

### CODE REVIEW

File: options/options.go [hunk 1]
```diff
@@ -76,6 +76,9 @@ type TerragruntOptions struct {
 	// packages can use the command without a direct reference back to the cli package (which would create a
 	// circular dependency).
 	RunTerragrunt func(*TerragruntOptions) error
+
+	// Keep the last found parent folder to be able to return it when get_parent_tfvars() is called from within the child
+	LastIncludedParentFolder string
```
Comment (brikis98) [inline options/options.go]: I'm not sure it makes sense to put this into `TerragruntOptions` and to modify at random points in the code. It's too hard to reason about.

We have the path of the parent folder while parsing the Terragrunt configuration. We should just save it! That is, update the [TerragruntConfig](https://github.com/gruntwork-io/terragrunt/blob/master/config/config.go#L20) struct with a `ParentPath` field. It should be set when we're working with the [Include](https://github.com/gruntwork-io/terragrunt/blob/master/config/config.go#L34) field in the `terragruntConfigFile` struct. 

After that, the `get_parent_tfvars()` could just return this field with no further work.

Comment (jocgir) [reply options/options.go]: It doesn't seem to be so simple. The problem is that we do not have the TerragruntConfig object when we evaluate the function.

Any suggestion.

Comment (brikis98) [reply options/options.go]: Actually, don't you have access to `include *IncludeConfig`? IIRC, if that isn't `nil`, it contains the path of the parent config file...

Comment (jocgir) [reply options/options.go]: In the context of the child, include is nil. It is the reason why this function is only working on the parent side and why I made this change to solve that problem.

Comment (brikis98) [reply options/options.go]: Ah, I gotcha.

I think the proper fix will require a bit of a refactor to the `parseConfigString` method. Right now, it tries to process interpolations in the string and _then_ parse it into a `terragruntConfigFile` struct. The problem is that to process those interpolations, you need info from that struct, such as any `include` setting.

So probably the better parsing approach would be to:

1. Parse the string into a struct that has just the top-level elements: `remote_state`, `include`, `terraform`. The contents of those elements are left "raw".

1. Resolve interpolations in each of those top-level elements, starting first with `include`, and make the struct available to the interpolation functions themselves.

**Pull request closed without merging**